cli/fathom_web/commands/label.py

```diff
@@ -0,0 +1,112 @@
+from html.parser import HTMLParser
+from functools import partial
+import multiprocessing
+import os
+import pathlib
+import re
+import shutil
+
+from click import argument, command, option, Path, STRING, progressbar
+
+
+@command()
+@option('--preserve-originals/--no-preserve-originals',
+        default=True,
+        help='Save original HTML files in a newly created `originals`'
+             ' directory in IN_DIRECTORY (default: True)')
+@argument('in_directory', type=Path(exists=True, file_okay=False))
+@argument('in_type', type=STRING)
+def main(in_directory, in_type, preserve_originals):
+    """
+    Add the ``data-fathom`` attribute with a value of IN_TYPE to the
+    opening tag of any ``<html>`` elements in the HTML pages in
+    IN_DIRECTORY. This tool is used to label an entire webpage (e.g.
+    IN_TYPE could be "article" for article webpages).
+    """
+    if preserve_originals:
+        originals_dir = pathlib.Path(in_directory) / 'originals'
+        try:
+            originals_dir.mkdir(parents=True)
+        except FileExistsError:
+            raise RuntimeError(f'Tried to make directory {originals_dir.as_posix()}, but it already exists. To protect'
+                               f' against unwanted data loss, please move or remove the existing directory.')
+    else:
+        originals_dir = None
+
+    list_of_items = os.listdir(in_directory)
+    number_of_items = len(list_of_items)
+
+    print_statements = [] # Capture any print statements to log at the end.
+
+    # Make a pool of workers. Each worker is in its own process. We use a scaling factor to account for the overhead of
+    # setting up all of the processes.
+    pool = multiprocessing.Pool(multiprocessing.cpu_count() * 4)
```

Comment:
Where did the 4 come from? My CPUs, at least, have 2 logical cores each.

Reply:
`cpu_count()` returns the number of logical cores, so the `4` is not the ratio of logical cores to physical cores. It is an overhead factor that mythmon suggested, since there is also work associated with setting up the IPC and for task switching.

I can make the number of workers an option; that's a good idea.

Reply:
I can actually remove the `4`, since my husband plotted the speed of the script as a function of number of processes, and beyond 8 or 9 showed little to no gains... Well, at least my hardware has 8 logical cores...